ReactQt/runtime/src/logger.cpp

```diff
@@ -0,0 +1,74 @@
+
+/**
+ * Copyright (c) 2017-present, Status Research and Development GmbH.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include "logger.h"
+#include <QBuffer>
+
+const QString LOG_SETTINGS_FILE = "logger_settings.ini";
+
+Logger& Logger::getInstance() {
+    static Logger logger;
+    return logger;
+}
+
+Logger::~Logger() {
+    delete m_settings;
+    m_settings = nullptr;
+}
+
+QString& Logger::devNull() {
+    return m_devnull;
+}
+
+const QSettings& Logger::settings() {
+    return *m_settings;
+}
+
+void Logger::registerModule(const QString& moduleName) {
+    if (!m_settings->contains(moduleName)) {
+        m_settings->setValue(moduleName, true);
+    }
+}
+
+void Logger::sync() {
+    m_settings->sync();
+}
+
+Logger::Logger() {
+    m_settings = new QSettings(LOG_SETTINGS_FILE, QSettings::IniFormat);
+
+    registerModule(UIMANAGER);
+    registerModule(FLEXBOX);
+    registerModule(VIEWMANAGER);
+}
+
+QDebug rnLog(const QString& module) {
+    Logger& logger = Logger::getInstance();
+    logger.devNull().clear();
+
+    if (!logger.settings().contains(module)) {
```

Comment:
curios, if `rnLog` is thread-safe method. At the moment we have: main thread, thread with JS engine communications, status-go thread in status-react rctstatus module, qtconcurrent calls from rctstatus module.

Reply:
I believe it should be ok:
- Mayers singleton is thread-safe in C++11 (https://stackoverflow.com/questions/1661529/is-meyers-implementation-of-the-singleton-pattern-thread-safe)
- QSettings methods are reentrant
- clearing of `m_devnull` has no sideeffects because data inside that string never used
- qDebug() also seems to be safe

Reply:
@vkjr thank you for checking!
P.S. When we install our custom handler for qDebug(), we should care about thread safety of handler implementation (at least that was the case with status-react).